src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1405,11 +1402,11 @@ def create_service_principal_for_rbac(
     import time
 
     graph_client = _graph_client_factory(cmd.cli_ctx)
-    role_client = _auth_client_factory(cmd.cli_ctx).role_assignments
 
-    if role and not scopes:
-        logger.warning(SCOPE_WARNING)
-        scopes = ['/subscriptions/' + role_client.config.subscription_id]
+    if role and not scopes or not role and scopes:
+        from azure.cli.core.azclierror import ArgumentUsageError
+        raise ArgumentUsageError("To create role assignments, "
+                                 "specify both --role and --scopes at the same time.")
```

Comment:
I think the last portion of the warning is unnecessary. It will be more intuitive if it looks like this:
```suggestion
        raise ArgumentUsageError("To create role assignments, "
                                 "specify both --role and --scopes.")
```